src/overlays/actors/ovl_En_Floormas/z_en_floormas.c

```diff
@@ -91,144 +141,1104 @@ static DamageTable D_808D38D4 = {
+        ((this->skelAnime.animPlaybackSpeed < 0.0f) && func_801378B8(&this->skelAnime, 6.0f))) {
+        if (this->actor.scale.x > 0.009f) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_FALL_WALK);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLOORMASTER_SM_WALK);
+        }
+    }
+
+    frame = this->skelAnime.animCurrentFrame;
+    if ((frame >= 7.0f) && (frame < 22.0f)) {
+        sp30 = Math_SinS(this->actor.shape.rot.y + 0x4268);
+        sp2C = Math_CosS(this->actor.shape.rot.y + 0x4268);
+        this->actor.shape.rot.y += this->unk_190;
+        this->actor.world.pos.x -= this->actor.scale.x * 2700.0f * (Math_SinS(this->actor.shape.rot.y + 0x4268) - sp30);
+        this->actor.world.pos.z -= this->actor.scale.x * 2700.0f * (Math_CosS(this->actor.shape.rot.y + 0x4268) - sp2C);
+    }
+}
+
+void func_808D1380(EnFloormas* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06009520, 3.0f, 0.0f, SkelAnime_GetFrameCount(&D_06009520.common), 2,
+                         -3.0f);
+    this->actor.speedXZ = 0.0f;
+    this->actor.gravity = 0.0f;
+    func_808D08D0(this);
+    func_800BBDAC(globalCtx, &this->actor, &this->actor.world.pos, 15.0f, 6, 20.0f, 300, 100, 1);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLOORMASTER_ATTACK);
+    this->actionFunc = func_808D1458;
+}
+
+void func_808D1458(EnFloormas* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808D161C(this);
+    }
+
+    this->actor.shape.rot.x += 0x140;
+    this->actor.world.pos.y += 10.0f;
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 0xAAA);
+    Math_StepToS(&this->unk_192, 0x4B0, 0x64);
+}
+
+void func_808D14DC(EnFloormas* this, GlobalContext* globalCtx) {
+    Vec3f sp34;
+    Vec3f sp28;
+
+    sp34.x = this->actor.world.pos.x;
+    sp34.z = this->actor.world.pos.z;
+    sp34.y = this->actor.floorHeight;
+
+    sp28.y = 2.0f;
+    sp28.x = Math_SinS(this->actor.shape.rot.y + 0x6000) * 7.0f;
+    sp28.z = Math_CosS(this->actor.shape.rot.y + 0x6000) * 7.0f;
+
+    func_800B1210(globalCtx, &sp34, &sp28, &D_801D15B0, 0x1C2, 0x64);
+    sp28.x = Math_SinS(this->actor.shape.rot.y - 0x6000) * 7.0f;
+    sp28.z = Math_CosS(this->actor.shape.rot.y - 0x6000) * 7.0f;
+    func_800B1210(globalCtx, &sp34, &sp28, &D_801D15B0, 0x1C2, 0x64);
+    func_800B9010(&this->actor, NA_SE_EN_FLOORMASTER_SLIDING - SFX_FLAG);
+}
+
+void func_808D161C(EnFloormas* this) {
+    this->unk_18E = 25;
+    this->actor.gravity = -0.15f;
+    this->actor.speedXZ = 0.5f;
+    this->actionFunc = func_808D1650;
+}
+
+void func_808D1650(EnFloormas* this, GlobalContext* globalCtx) {
+    f32 temp_f0_2;
+
+    if (this->unk_18E != 0) {
+        this->unk_18E--;
+    }
+
+    Math_StepToF(&this->actor.speedXZ, 15.0f, SQ(this->actor.speedXZ) * 0.33333334f);
```

Comment:
```suggestion
    Math_StepToF(&this->actor.speedXZ, 15.0f, SQ(this->actor.speedXZ) * (1.0f / 3.0f));
```
Similar idea to the previous comment